src/main/java/com/example/ludogorieSoft/village/controllers/AdministratorController.java

```diff
@@ -15,10 +16,12 @@
 
 @RestController
 @RequestMapping("/api/v1/admins")
-@AllArgsConstructor
+@RequiredArgsConstructor
 public class AdministratorController {
```

Comment:
This class has too many methods now and does not follow the Single Responsibility Principle. Would be better to separate it to at least 2 classes: one for the admins CRUD and another one for villages CRUD.

Reply:
Yes, you are right. At first there weren't that many, but method after method the class got bigger. I left it like this because these are admin functions, but it might be good to separate it.